files/zh-cn/web/javascript/reference/global_objects/promise/index.md

```diff
@@ -99,148 +115,136 @@ const promiseC = promiseA.then(handleFulfilled2, handleRejected2);
+promiseA.then((val) => console.log("异步日志记录有值：", val));
+console.log("立即记录");
 
-// produces output in this order:
-// immediate logging
-// asynchronous logging has val: 777
+// 按以下顺序产生输出：
+// 立即记录
+// 异步日志记录有值：777
 ```
 
-### 追踪现有设置对象
+### Thenable
 
-设置对象（settings object）是 JavaScript 代码运行时用于提供附加信息的[环境](https://html.spec.whatwg.org/multipage/webappapis.html#environment-settings-object)。它包含了领域（realm）和模块映射（module map），以及 HTML 的特定信息，如来源（origin）等。对现有设置对象的追踪保证了浏览器知道用户给定的哪些代码片段需要使用。
+在 Promise 成为 JavaScript 语言的一部分之前，JavaScript 生态系统已经实现了多个 Promise 实现。尽管它们在内部的表示方式不同，但至少所有类 Promise 的对象都实现了 _Thenable_ 接口。一个 thenable 对象实现了 [`.then()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise/then) 方法，该方法被调用时需要传入两个回调函数，一个用于 Promise 被兑现时调用，一个用于 Promise 被拒绝时调用。Promise 也是 thenable 对象。
 
-为了更好地说明这一点，我们在这里进一步探讨领域是如何引发问题的。我们可以粗略地认为**领域**是一个全局对象。其独特之处在于，它拥有运行 JavaScript 代码所需的所有信息。这包括像 [`Array`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array) 和 [`Error`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Error) 这样的对象。每一个设置对象都有自己的“副本”，而且它们与副本之间是不共享的。这可能会导致一些与 promise 相关的意外行为。为了解决这个问题，我们需要追踪**现有设置对象**（incumbent settings object）。它表示负责用户某个函数调用工作的特定信息。
-
-我们可以尝试在文档中嵌入 [`<iframe>`](/zh-CN/docs/Web/HTML/Element/iframe)，并让其与父级上下文通信。由于所有的 web API 都有现有设置对象，下面的代码能够在所有的浏览器中运行：
+为了与现有的 Promise 实现进行交互，JavaScript 语言允许在 Promise 的位置使用 thenable 对象。例如，[`Promise.resolve`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve) 方法不仅可以解析 Promise 对象，还可以追踪 thenable 对象。
 
-```html
-<!DOCTYPE html>
-<iframe></iframe> <!-- we have a realm here -->
-<script> // we have a realm here as well
-  const bound = frames[0].postMessage.bind(
-    frames[0], "some data", "*");
-    // bound is a built-in function -- there is no user
-    // code on the stack, so which realm do we use?
-  window.setTimeout(bound);
-  // this still works, because we use the youngest
-  // realm (the incumbent) on the stack
-</script>
+```js
+const aThenable = {
+  then(onFulfilled, onRejected) {
+    onFulfilled({
+      // thenable 对象被兑现为另一个 thenable 对象
+      then(onFulfilled, onRejected) {
+        onFulfilled(42);
+      },
+    });
+  },
+};
+
+Promise.resolve(aThenable); // 一个兑现值为 42 的 Promise
 ```
 
-同样的概念也适用与 promise。如果我们稍加修改上面的示例，我们就能得到这个：
+### Promise 并发
 
-```html
-<!DOCTYPE html>
-<iframe></iframe> <!-- we have a realm here -->
-<script> // we have a realm here as well
-  const bound = frames[0].postMessage.bind(
-    frames[0], "some data", "*");
-    // bound is a built in function -- there is no user
-    // code on the stack -- which realm do we use?
-  Promise.resolve(undefined).then(bound);
-  // this still works, because we use the youngest
-  // realm (the incumbent) on the stack
-</script>
-```
+`Promise` 类提供了四个静态方法来促进异步任务的[并发](https://zh.wikipedia.org/wiki/并发计算)：
 
-如果我们修改代码，使用文档中的 `<iframe>` 来监听发送的消息，我们可以观察到现有设置对象的影响：
+- {{jsxref("Promise.all()")}}
+  - : 当**所有**的 Promise 都被兑现时，`Promise.all` 方法返回一个成功兑现的 Promise；当**任意一个** Promise 被拒绝时，`Promise.all` 方法返回一个被拒绝的 Promise。
```

Comment:
The description is not correct. The method would return a promise _even if_ all the passed promises is not settled.